subprojects/core/src/main/java/org/gradle/execution/taskgraph/DefaultTaskExecutionGraph.java

```diff
@@ -152,8 +153,14 @@ public void execute(Collection<? super Throwable> failures) {
         } else if (!graphListeners.isEmpty()) {
             LOGGER.warn("Ignoring listeners of task graph ready event, as this build (" + gradleInternal.getIdentityPath() + ") has already executed work.");
         }
+
         try {
-            planExecutor.process(executionPlan, failures, new BuildOperationAwareExecutionAction(nodeExecutors, buildOperationExecutor.getCurrentOperation()));
+            planExecutor.process(executionPlan, failures,
+                new BuildOperationAwareExecutionAction(
+                    buildOperationExecutor.getCurrentOperation(),
+                    new InvokeNodeExecutorsAction(nodeExecutors, new ProjectExecutionServiceRegistry())
```

Comment:
Why did you split this up?

Reply:
I made a few changes back and forth and ended up not altering the behavior, but it seemed like a good change to have them in two separate types, so I did not revert the split.